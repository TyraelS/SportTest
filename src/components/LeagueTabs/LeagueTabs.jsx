import React from 'react';

import LeagueTabsWrapper from './LeagueTabsWrapper.style';
import LeagueTab from 'components/LeagueTab';

const LeagueTabs = ({ leagues }) => {
  return (
    <LeagueTabsWrapper>
      {leagues.entrySeq().map(([key, league]) => (
        <LeagueTab
          key={key}
          name={league.get('name', '')}
          country={league.get('countryCode', '')}
        >
          {league.get('name', '')}
        </LeagueTab>
      ))}
    </LeagueTabsWrapper>
  );
};

LeagueTabs.displayName = 'LeagueTabs';

export default LeagueTabs;
