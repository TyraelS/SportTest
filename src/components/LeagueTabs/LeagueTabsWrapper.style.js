import styled from 'styled-components';

const LeagueTabsWrapper = styled.div`
  width: 65%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  overflow-y: scroll;
  color: ${props => props.theme.textLeagues};
  background-color: ${props => props.theme.leaguesItemEven};
  > div {
    border: 1px solid ${props => props.theme.leaguesItemBorder};
    border-bottom: none;
    &:last-child {
      border-bottom: 1px solid ${props => props.theme.leaguesItemBorder};
    }
  }
  > div:nth-child(odd) {
    background-color: ${props => props.theme.leaguesItemOdd};
  }
`;

export default LeagueTabsWrapper;
