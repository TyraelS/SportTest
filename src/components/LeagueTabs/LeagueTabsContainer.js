import { compose, setDisplayName } from 'recompose';
import { Map } from 'immutable';
import { connect } from 'react-redux';

import LeagueTabs from './LeagueTabs';

export const mapStateToProps = state => ({
  leagues: state.get('leagues', Map())
});

export default compose(
  setDisplayName('LeagueTabsContainer'),
  connect(mapStateToProps)
)(LeagueTabs);
