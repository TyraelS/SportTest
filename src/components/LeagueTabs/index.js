import LeagueTabsContainer from './LeagueTabsContainer';

export default LeagueTabsContainer;
