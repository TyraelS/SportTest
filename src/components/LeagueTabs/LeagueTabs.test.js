import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import LeagueTabs from './LeagueTabs';

describe('Given the LeagueTabs component with provided props', () => {
  const leagues = fromJS({
    SBTC_01: {
      key: 'SBTC_01',
      name: 'SortofLeague',
      countryCode: 'GBR'
    },
    SBTC_02: {
      key: 'SBTC_02',
      name: 'AnotherLeague',
      countryCode: 'USA'
    }
  });

  describe('when the component is rendered', () => {
    const component = shallow(<LeagueTabs leagues={leagues} />);

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
