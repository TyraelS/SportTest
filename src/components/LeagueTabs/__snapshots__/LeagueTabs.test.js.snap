// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the LeagueTabs component with provided props when the component is rendered should match the snapshot 1`] = `
<ForwardRef(styled.div)>
  <LeagueTab
    country="GBR"
    key="SBTC_01"
    name="SortofLeague"
  >
    SortofLeague
  </LeagueTab>
  <LeagueTab
    country="USA"
    key="SBTC_02"
    name="AnotherLeague"
  >
    AnotherLeague
  </LeagueTab>
</ForwardRef(styled.div)>
`;
